### PR TITLE
Refactored merchant_items and merchant controller to give us 100% and added tests for them

### PR DESCRIPTION
**Pull request by MDelarosa1993**
1. MerchantItemsController:
- Removed redundant merchant rendering in the show action.

2. MerchantsController:
- Added a find action that searches for a merchant by name, with proper error handling for missing parameters and case where no merchant is found.
- Updated filter_by_name method to include case-insensitive search and return only the first matching merchant.

3. Model Updates:
- Enhanced the filter_by_name method to handle name filtering with case insensitivity.

4. Tests:
- Expanded test coverage for the find action, including scenarios for multiple matches, exact matches, and missing parameters.
- Overall, these changes improve the handling of merchant searches and enhance error responses.

### CODE REVIEW

File: spec/requests/api/v1/merchants_requests_spec.rb [hunk 1]
```diff
@@ -208,5 +208,37 @@
...
+
+  describe 'GET #find' do
+    before do
+      Merchant.create!(name: "Ring World")
+      Merchant.create!(name: "Turing")
+      Merchant.create!(name: "Alpha")
+    end
+
+    it 'returns the first merchant alphabetically when multiple matches are found' do
+      get "/api/v1/merchants/find", params: { name: "Ring" }
```
Comment (litobot) [inline spec/requests/api/v1/merchants_requests_spec.rb:222]: I didn't know you could also do it like this.

This is the same as:

get "/api/v1/merchants/find?name=ring"

Correct?

**Review by litobot — APPROVED** [review]
This looks awesome, Melchor.  You taught me a couple of things as well.  I'm gonna have to do some digging to find out more.

**Comment by jimmacur** [comment]
“Nice update to handle missing parameters and edge cases more gracefully. The added error handling with ArgumentError in find is a great improvement, and the tests ensure robustness.
